main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os/user"
+	"strings"
+	"sync"
+
+	"github.com/DeFacto-Team/factom-network-restart/config"
+	"github.com/DeFacto-Team/factom-network-restart/portainer"
+	log "github.com/sirupsen/logrus"
+)
+
+func main() {
+
+	var err error
+	var live bool
+	var conf *config.Config
+
+	log.SetLevel(5)
+
+	usr, err := user.Current()
+	if err != nil {
+		log.Error(err)
+	}
+
+	// default config location
+	configFile := usr.HomeDir + "/.factom-restart/config.yaml"
+
+	// check if custom config location passed as flag
+	flag.StringVar(&configFile, "c", configFile, "config.yaml path")
+
+	// parse dry run flag
+	flag.BoolVar(&live, "live", live, "live bool")
+
+	flag.Parse()
+
+	log.Info("Using config: ", configFile)
+
+	// load config
+	if conf, err = config.NewConfig(configFile); err != nil {
+		log.Fatal(err)
+	}
+
+	log.Info("Starting restart system")
+	log.Info("Portainer endpoint: ", conf.Endpoint)
+
+	if live {
+		log.Warn("LIVE mode: network will be restarted!")
+	} else {
+		log.Info("DRY RUN mode: simulating restart")
+	}
+
+	// initialize portainer
+	p := portainer.NewPortainer(conf.Username, conf.Password, conf.Endpoint)
+
+	// get all endpoints from portainer
+	endpoints, err := p.GetSwarmEndpoints()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// get factomd containers for each endpoint
+	for i := range endpoints {
```

Comment:
It's much more readable here and in the loop below to just iterate through the elements themselves rather than use the index imo.

Reply:
fixed 94928b00b692f8bcc76bb193447507f9262c259c